internal/codeintel/codenav/shared/types.go

```diff
@@ -54,3 +56,25 @@ type Position struct {
 	Line      int
 	Character int
 }
+
+func (p Position) Compare(other Position) int {
+	return cmp.Or(cmp.Compare(p.Line, other.Line), cmp.Compare(p.Character, other.Character))
```

Comment:
```suggestion
	if ret := cmp.Compare(p.Line, other.Line); ret != 0 {
		return ret
	}
	return cmp.Compare(p.Character, other.Character)
```

The current code will do a redundant comparison (this is what I was talking about earlier: https://github.com/sourcegraph/sourcegraph/pull/63029#issuecomment-2144984378)